yogstation/code/modules/gauntlet/gauntlet.dm

```diff
@@ -0,0 +1,611 @@
+	var/obj/item/infinity_stone/S = I
+	if(istype(I, /obj/item/infinity_stone))
+		add_stone(S.gem_type, user)
+		qdel(I)
+		user.dropItemToGround(src)
+		to_chat(user, "<span class='warning'>The gauntlet becomes too hot to touch.</span>")
+
+////obj/item/t_scanner/attack_self(mob/user)
+
+/obj/item/clothing/gloves/gauntlet/proc/snap(user)
+	if(applied_gems.len == 6)
+		to_chat(user, "<span class='warning'>You destroy humanity</span>")
+		for(var/mob/living/carbon/M in GLOB.mob_list)
+			if(M.stat != DEAD)
+				if(!(user == M))
+					M.gib()
+	else
+		to_chat(user, "<span class='warning'>You need all the infinity gems to destroy humanity.</span>")
+
+/obj/item/clothing/gloves/gauntlet/proc/power_stone_activate()
+	var/obj/effect/proc_holder/spell/targeted/tesla/gauntlet/J = new /obj/effect/proc_holder/spell/targeted/tesla/gauntlet()
+	spells_list += J
+
+/obj/item/clothing/gloves/gauntlet/proc/space_stone_activate() // Now your body is space protected!!
+	var/obj/effect/proc_holder/spell/targeted/ethereal_jaunt/gauntlet/J = new /obj/effect/proc_holder/spell/targeted/ethereal_jaunt/gauntlet()
+	spells_list += J
+
+/obj/item/clothing/gloves/gauntlet/proc/reality_stone_activate() // You have a fist of steel!
+	var/obj/effect/proc_holder/spell/spacetime_dist/gauntlet/J = new /obj/effect/proc_holder/spell/spacetime_dist/gauntlet()
+	spells_list += J
+
+/obj/item/clothing/gloves/gauntlet/proc/soul_stone_activate()
+	var/obj/effect/proc_holder/spell/aimed/resurrect/J = new /obj/effect/proc_holder/spell/aimed/resurrect()
+	spells_list += J
+
+/obj/item/clothing/gloves/gauntlet/proc/time_stone_activate()
+	var/obj/effect/proc_holder/spell/aimed/time/J = new /obj/effect/proc_holder/spell/aimed/time()
+	spells_list += J
+
+/obj/item/clothing/gloves/gauntlet/proc/mind_stone_activate()
+	var/obj/effect/proc_holder/spell/aimed/animate/J = new /obj/effect/proc_holder/spell/aimed/animate()
+	spells_list += J
+
+/obj/item/clothing/gloves/gauntlet/proc/add_stone(New_stone, user)
+	for(var/D in applied_gems)
+		if(New_stone == D || New_stone > 6 || New_stone < 1)
+			to_chat(user, "You've broken something.")
+			return
+	var/image/O = image(icon = 'yogstation/icons/obj/gauntlet.dmi', icon_state = "stone_[New_stone]")
+	add_overlay(O)
+	applied_gems += New_stone
+	to_chat(user, "You add the stone to the [src].")
+	if(New_stone == 1)
+		power_stone_activate()
+	if(New_stone == 2)
+		space_stone_activate()
+	if(New_stone == 3)
+		reality_stone_activate()
+	if(New_stone == 4)
+		soul_stone_activate()
+	if(New_stone == 5)
+		time_stone_activate()
+	if(New_stone == 6)
+		mind_stone_activate()
+
+
+/obj/item/clothing/gloves/gauntlet/proc/remove_stone(Old_stone) //Not used right now and not fully implemented
+	cut_overlays()
+	for(var/N in applied_gems)
+		var/image/O = image(icon = 'yogstation/icons/obj/gauntlet.dmi', icon_state = "stone_[N]")
+		add_overlay(O)
+
+//Spells
+
+/obj/effect/proc_holder/spell/targeted/tesla/gauntlet
```

Comment:
a bunch of these spells have variables that are identical to the ones they inherit, like the name and description, so they don't need to be here as well.